utils/schemed-kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/schemedkafka/schemaregistry/universal/ParsedSchemaSupport.scala

```diff
@@ -61,6 +60,15 @@ class AvroSchemaSupport(kafkaConfig: KafkaConfig) extends ParsedSchemaSupport[Av
 
   override def typeDefinition(schema: ParsedSchema): TypingResult = AvroSchemaTypeDefinitionExtractor.typeDefinition(schema.cast().rawSchema())
 
+  def extractParameters(schema: ParsedSchema)(implicit nodeId: NodeId): ValidatedNel[ProcessCompilationError, List[Parameter]] = {
```

Comment:
Now it is easier to see what is duplicated across those json/avro support classes. I would like to have as few methods as possible in this interface to be easy to recognize how  json/avro implementations are different from each other.  IMO we have to :
- rename `SinkValueParameter` to sth like ` SchemaBasedParameter` or `SchemaFormParameter` to be clear that it is not only used in sinks, but also in sources - everwhere where you have a schema and want to have form based on this
- rename `extractSinkValueParameter` to just `extractParameter`
- make `extractParameters` return  `SchemaBasedParameter`/`SchemaFormParameter` and invoke one from each other so it will be visible that it is just an overloaded method with some default values for `rawMode`

Reply:
I've renamed `SinkValueParameter` to `SchemaBasedParameter` as well as `extractSinkValueParameter` to `extractParameter`. Then I've implemented inside `UniversalSchemaSupport` both `extractParameters` and `prepareMessageFormatter` based on methods from inside that trait.

Reply:
Thanks, refactored

Reply:
Wait, I will move it to schema package